bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/handler/GenericConfiguration.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.generic.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+
+/**
+ * The {@link GenericMQTTThingHandler} manages Things that are responsible for MQTT components.
+ * This class contains the necessary configuration for such a Thing handler.
+ *
+ * @author Jochen Klein - Initial contribution
+ */
+@NonNullByDefault
+public class GenericConfiguration {
```

Comment:
I'm not sure if I like the class name. Could be generic thing config, generic binding config and so on. A better name is probably GenericThingConfiguration.